### PR TITLE
Multiple datasources example

### PR DESCRIPTION
**Pull request by jlmayorga**
New example added supporting multiple datasources with the S2I image and the WildFly Helm chart

### CODE REVIEW

File: examples/postgresql-multiple-datasources/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+* Helm chart for WildFly `wildfly/wildfly`. Minimal version `2.0.0`.
+
+# WildFly image API
+Environment variables from the [WildFly image API](https://github.com/wildfly/wildfly-cekit-modules/blob/main/jboss/container/wildfly/run/api/module.yaml) that must be set in the OpenShift deployment environment
+
+* NONE
+
+# WildFly cloud feature-pack added features
+
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/README.md:40]: The env variables related to datasource configuration in the helm chart file (eg: DB_SERVICE_PREFIX_MAPPING, PREFIX_JNDI, ... ) should be listed there. Yo ucould add a link to the env variable doc: https://github.com/wildfly/wildfly-cekit-modules/blob/main/jboss/container/wildfly/launch/datasources/module.yaml

File: examples/postgresql-multiple-datasources/helm.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+build:
+  uri: https://github.com/wildfly/wildfly-s2i
+  mode: s2i
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/helm.yaml:3]: Our existing examples need some cleanup. We could cleanup this new one. you can remove this field, that is the default value.

File: examples/postgresql-multiple-datasources/helm.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  contextDir: examples/postgresql-multiple-datasources
+  ref: main
+  s2i:
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/helm.yaml:6]: s2i is the default mode, this field can be remove as well as version, builderImage and runtimeIme.

File: examples/postgresql-multiple-datasources/helm.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    version: latest
+    builderImage: quay.io/wildfly/wildfly-s2i-jdk11
+    runtimeImage: quay.io/wildfly/wildfly-runtime-jdk11
+deploy:
+  replicas: 1
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/helm.yaml:11]: Value of 1 is the default, you can remove this field.

File: examples/postgresql-multiple-datasources/pom.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    <name>WildFly S2I - Multiple PostgreSQL Datasource </name>
+    <description>An application that connects to multiple PostgreSQL databases</description>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <version.maven.war.plugin>3.3.2</version.maven.war.plugin>
+        <version.wildfly>26.1.1.Final</version.wildfly>
+        <version.wildfly.plugin>4.0.0.Alpha2</version.wildfly.plugin>
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/pom.xml:17]: You could use version 4.0.0.Beta3.

File: examples/postgresql-multiple-datasources/pom.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    <groupId>org.wildfly.wildfly-s2i.examples</groupId>
+    <artifactId>postgresql-multiple-datasources</artifactId>
+    <version>1.0</version>
+    <packaging>war</packaging>
+
+    <name>WildFly S2I - Multiple PostgreSQL Datasource </name>
+    <description>An application that connects to multiple PostgreSQL databases</description>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/pom.xml:13]: You should use version 11.

File: examples/postgresql-multiple-datasources/pom.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/pom.xml:14]: You should user version 11

File: examples/postgresql-multiple-datasources/pom.xml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        <version.maven.war.plugin>3.3.2</version.maven.war.plugin>
+        <version.wildfly>26.1.1.Final</version.wildfly>
+        <version.wildfly.plugin>4.0.0.Alpha2</version.wildfly.plugin>
+        <version.wildfly.cloud.galleon.pack>1.1.2.Final</version.wildfly.cloud.galleon.pack>
+        <version.wildfly.datasources.galleon.pack>2.2.1.Final</version.wildfly.datasources.galleon.pack>
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/pom.xml:19]: You could use version 2.2.4 (latest release).

File: examples/postgresql-multiple-datasources/src/main/java/org/wildfly/plugins/demo/tasksrs/model/TaskDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ */
+@RequestScoped
+public class TaskDaoImpl implements TaskDao {
+
+    @PersistenceContext(type = PersistenceContextType.EXTENDED,unitName = "primary")
+    private EntityManager em;
+
+    EntityManager getEM() {
+//        if(em == null) {
```
Comment (jfdenise) [inline examples/postgresql-multiple-datasources/src/main/java/org/wildfly/plugins/demo/tasksrs/model/TaskDaoImpl.java:40]: You could perhaps remove this comment.

**Review by jfdenise — CHANGES_REQUESTED** [review]
Thank-you very much for your contribution. That is great! I added few comments.

File: examples/postgresql-multiple-datasources/README.md (hunk 1 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/README.md:53]: Added information about environment variables and linked to official documentation.

File: examples/postgresql-multiple-datasources/helm.yaml (hunk 2 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/helm.yaml]: Done, removed these items as they are the default values.

File: examples/postgresql-multiple-datasources/helm.yaml (hunk 3 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/helm.yaml]: Done, removed these items as they are the default values.

File: examples/postgresql-multiple-datasources/helm.yaml (hunk 4 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/helm.yaml]: Done, removed these items as they are the default values.

File: examples/postgresql-multiple-datasources/pom.xml (hunk 5 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/pom.xml]: Updated with latest available version (4.0.0.Beta3)

File: examples/postgresql-multiple-datasources/pom.xml (hunk 6 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/pom.xml]: Updated to use version 11

File: examples/postgresql-multiple-datasources/pom.xml (hunk 7 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/pom.xml]: Updated to use version 11

File: examples/postgresql-multiple-datasources/pom.xml (hunk 8 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/pom.xml]: Updated to use latest version (2.2.4.Final)

File: examples/postgresql-multiple-datasources/src/main/java/org/wildfly/plugins/demo/tasksrs/model/TaskDaoImpl.java (hunk 9 above)
Comment (jlmayorga) [reply examples/postgresql-multiple-datasources/src/main/java/org/wildfly/plugins/demo/tasksrs/model/TaskDaoImpl.java]: Removed commented code section

**Comment by jlmayorga** [comment]
@jfdenise, Thanks for your input! I addressed your comments and pushed the latest updates

**Comment by jlmayorga** [comment]
@jfdenise, commits squashed. Thank you!

**Comment by jfdenise** [comment]
Merged, thank-you!

**Pull request merged**